src/main/scala/firrtl/ir/IR.scala

```diff
@@ -448,9 +473,16 @@ case class Module(info: Info, name: String, ports: Seq[Port], body: Statement) e
 /** External Module
   *
   * Generally used for Verilog black boxes
+  * @param defname Defined name of the external module (ie. the name Firrtl will emit)
   */
-case class ExtModule(info: Info, name: String, ports: Seq[Port]) extends DefModule {
-  def serialize: String = serializeHeader("extmodule")
+case class ExtModule(
+    info: Info,
+    name: String,
+    ports: Seq[Port],
+    defname: String,
```

Comment:
The motivation behind defname is that, from the perspective of Firrtl, blackboxes with different parameters that affect the IO are different external modules. We discussed a couple of ways to get around this. One way would have been to essentially implement Verilog parameterization in Firrtl. The alternative we chose was just to let the user override the blackbox module name via defname. Thus, "name" is used in the Firrtl netlist, and defname is what we actually emit.